crates/editor_ui/src/meshless_visualizer.rs

```diff
@@ -1,72 +1,194 @@
-                material: handle.mat.clone(),
-                transform: *transform,
-                ..default()
-            })
-            .insert(RenderLayers::layer((RenderLayers::TOTAL_LAYERS - 1) as u8));
+// this will need to be changed for billboarding next
+pub fn rotate_icons(
+    mut icons: Query<&mut Transform, (With<Sprite3dComponent>, Without<EditorCameraMarker>)>,
+    editor_camera: Query<&Transform, (With<EditorCameraMarker>, Without<Sprite3dComponent>)>,
+) {
+    let Ok(cam) = editor_camera.get_single() else {
+        return;
+    };
+    for mut icon in icons.iter_mut() {
+        icon.look_at(cam.translation, Vec3::Y);
     }
 }
 
-/// Creates the default mesh to use
-pub fn setup_meshless_mesh(
-    mut meshes: ResMut<Assets<Mesh>>,
-    mut handle: ResMut<MeshlessMeshMat>,
-    mut mats: ResMut<Assets<StandardMaterial>>,
+// TODO: update this to follow the new method so that either a mesh or a 3d sprite can be added to whatever
+// a user wants
+
+/// This will create a way to have any entity with CustomMeshlessMarker have a way to be visualized by the user
+/// Additionally, the user can either choose their own mesh and material to use or default to the white sphere
+pub fn visualize_custom_meshless(
+    mut commands: Commands,
+    ass: Res<AssetServer>,
+    objects: Query<(
+        Entity,
+        &Transform,
+        &CustomMeshless,
+        Option<&Handle<Mesh>>,
+        Option<&Handle<StandardMaterial>>,
+    )>,
 ) {
-    handle.mesh = meshes.add(
-        shape::UVSphere {
-            radius: 0.5,
-            ..default()
+    /* NOTES: Maybe this should instead of a struct that is a component, there should be a trait that
+            can be impl'd such that the user can pair up different Components and meshes that this function then handles overall.
+            An example could be that `objects` is (Entity, &Transform) With<impl CustomMeshless> so then anything that impls it by the user can
+            be visualized via their impl, otherwise should be defaulted (or derived default if need be).
+    */
+
+    for (entity, transform, custom, mesh, mat) in objects.iter() {
+        match (mesh, mat) {
+            (Some(_), Some(_)) => {}
+            _ => {
+                commands.entity(entity).insert((
+                    // NOTE: 2d case is not currently covered
+                    MaterialMeshBundle {
+                        mesh: ass.add(custom.visual.mesh.clone()),
+                        material: ass.add(custom.visual.material.clone()),
+                        transform: *transform,
+                        ..default()
+                    },
+                    RenderLayers::layer((RenderLayers::TOTAL_LAYERS - 1) as u8),
+                ));
+            }
         }
-        .into(),
-    );
-    handle.mat = mats.add(StandardMaterial::default());
+    }
+}
+
+/// loads the icons for the different types of lights and camera
+pub fn load_light_icons(
```

Comment:
Maybe Bevy Asset loader can help a bit here

Reply:
Yes I will be adding that with a commit here soon!